test/Interop/Cxx/class/safe-interop-mode.swift

```diff
@@ -0,0 +1,29 @@
+
+// RUN: rm -rf %t
+// RUN: split-file %s %t
+// RUN: %target-swift-frontend -typecheck -verify -I %swift_src_root/lib/ClangImporter/SwiftBridging  -I %t/Inputs  %t/test.swift -enable-experimental-feature AllowUnsafeAttribute -enable-experimental-feature WarnUnsafe -enable-experimental-feature NonescapableTypes -cxx-interoperability-mode=default -cxx-safe-interop -diagnostic-style llvm 2>&1
+
+//--- Inputs/module.modulemap
+module Test {
+    header "nonescapable.h"
+    requires cplusplus
+}
+
+//--- Inputs/nonescapable.h
+#include "swift/bridging"
+
+struct SWIFT_NONESCAPABLE View {};
+
+struct SWIFT_ESCAPABLE Owner {};
+
+struct Unannotated {};
+
+//--- test.swift
+
+import Test
+
+func useUnsafeParam(x: Unannotated) { // expected-warning{{reference to unsafe struct 'Unannotated'}}
+}
+
+func useSafeParams(x: Owner) {
```

Comment:
I cannot really add a test for view yet, because I'd get errors about the compiler not being able to infer lifetimes. I'd actually need the lifetimebound patch landed before I could add such tests.

Reply:
Two options if you want to get the test in place now:
1. If the `@lifetime` attribute spelling patch has landed, `__attribute__((swift_attr("@lifetime(...)")))` should work
2. You could put "// expected-error{...}" for all of the errors you get today, and remove them once lifetime inference is there

Reply:
Thanks for the tips! Option 1 worked! Unfortunately, there is no way to make option 2 work at the moment because there is a bug with the diagnostics, it does not have a source location:

```
<unknown>:0: error: unexpected error produced: cannot infer lifetime dependence, no parameters found that are either ~Escapable or Escapable with a borrowing ownership
```



This is something that I plan to address separately in a different PR at some point.

Reply:
The compiler has an option `-verify-ignore-unknown` to ignore diagnostics without source location information. We don't love using it, but it's here for this purpose. The `@lifetime` solution is better, though!